src/controls/mdfbChips.js

```diff
@@ -28,16 +28,32 @@ module.exports = function ($timeout) {
       if (scope.field.transformFunc) {
         var varToWatch = 'form.' + scope.field.name + '.$viewValue'
         scope.$watch(varToWatch, function (value, oldValue) {
+          var promises = []
           for (var i = 0; i < value.length; i++) {
             if (!value[i].text) {
               var code = angular.copy(value[i].code)
-              value[i].text = getSetOptionResult(scope.field.name, i, code)
+              value[i].text = getSetOptionResult(scope.field.name, i, code, promises)
             }
           }
+          Promise.all(promises).then(function () {
+            var fieldIsValid = true
+            for (var i = 0; i < value.length; i++) {
```

Comment:
This is minor, but: a better way to do this might to be use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some